web/src/installers/index.ts

```diff
@@ -1278,6 +1346,34 @@ export class Installer {
         return {error: {message: `Prometheus service types are supported for version "0.48.1-16.10.0" and later, not "${this.spec.prometheus.version}"`}};
       }
     }
+
+    // K3s is only compatible with a subset of addons
+    if (this.spec.k3s && this.spec.k3s.version) {
+      const compatibleAddons = Installer.getK3sCompatibleAddons();
+      const incompatibleAddons: string[] = [];
+      _.each(specSchema.properties, (val, key) => {
+        if (key !== "k3s" && this.spec[key] && this.spec[key].version && compatibleAddons.indexOf(key) == -1) {
```

Comment:
I might have chosen [.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) here instead of indexOf